libpoolprotocols/PoolURI.cpp

```diff
@@ -0,0 +1,98 @@
+#include <libpoolprotocols/PoolURI.h>
+#include <map>
+
+using namespace dev;
+using namespace std;
+
+static void toLower(string &str)
+{
+	string::iterator end = str.end();
+	for (string::iterator it = str.begin(); it != end; it++)
+		*it = tolower(*it);
+}
+
+typedef struct {
+	ProtocolFamily family;
+	SecureLevel secure;
+	unsigned version;
+} SchemeType;
+
+static map<string, SchemeType> s_schemes = {
+	{"stratum",	{ProtocolFamily::STRATUM, SecureLevel::NONE,  0}},
+	{"stratum1",	{ProtocolFamily::STRATUM, SecureLevel::NONE,  1}},
+	{"stratum2",	{ProtocolFamily::STRATUM, SecureLevel::NONE,  2}},
+	{"stratum+tcp",	{ProtocolFamily::STRATUM, SecureLevel::NONE,  0}},
+	{"stratum1+tcp",{ProtocolFamily::STRATUM, SecureLevel::NONE,  1}},
+	{"stratum2+tcp",{ProtocolFamily::STRATUM, SecureLevel::NONE,  2}},
+	{"stratum+tls",	{ProtocolFamily::STRATUM, SecureLevel::TLS12, 0}},
+	{"stratum1+tls",{ProtocolFamily::STRATUM, SecureLevel::TLS12, 1}},
+	{"stratum2+tls",{ProtocolFamily::STRATUM, SecureLevel::TLS12, 2}},
+	{"stratum+ssl",	{ProtocolFamily::STRATUM, SecureLevel::TLS12, 0}},
+	{"stratum1+ssl",{ProtocolFamily::STRATUM, SecureLevel::TLS12, 1}},
+	{"stratum2+ssl",{ProtocolFamily::STRATUM, SecureLevel::TLS12, 2}},
+	{"http",	{ProtocolFamily::GETWORK, SecureLevel::NONE,  0}},
+	{"https",	{ProtocolFamily::GETWORK, SecureLevel::TLS12, 0}},
```

Comment:
same here, also i would not support secure connections at the beginning so either comment it out or we need some not yet supported warning there once ``-P`` is supported.